### PR TITLE
fix: postgres ssh connection errors

### PR DESCRIPTION
**Pull request by rephus**
Closes: 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 1]
```diff
@@ -169,8 +169,25 @@ export class PostgresClient<
             fields: Record<string, { type: DimensionType }>;
             rows: Record<string, any>[];
         }>((resolve, reject) => {
-            pool = new pg.Pool(this.config);
+            pool = new pg.Pool({
+                ...this.config,
+                connectionTimeoutMillis: 5000,
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:174]: Fixes https://lightdash.slack.com/archives/C02GQKJK84Q/p1699002194304279?thread_ts=1698961392.481489&cid=C02GQKJK84Q

image

Comment (rephus) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:174]: image

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 2]
```diff
@@ -169,8 +169,25 @@ export class PostgresClient<
...
+            });
+
+            pool.on('error', (err) => {
+                console.error(`Postgres pool error ${err.message}`);
+                reject(err);
+            });
 
+            pool.on('connect', (_client: pg.PoolClient) => {
+                // On each new client initiated, need to register for error(this is a serious bug on pg, the client throw errors although it should not)
+                _client.on('error', (err: Error) => {
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:184]: Fixes 
image https://github.com/brianc/node-postgres/issues/2439#issuecomment-757691278

Comment (rephus) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:184]: image

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 3]
```diff
@@ -183,6 +200,12 @@ export class PostgresClient<
                     return;
                 }
 
+                client.on('error', (e) => {
+                    console.error(`Postgres client error ${e.message}`);
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:204]: Captures 
image

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 4]
```diff
@@ -169,8 +169,25 @@ export class PostgresClient<
...
             rows: Record<string, any>[];
         }>((resolve, reject) => {
-            pool = new pg.Pool(this.config);
+            pool = new pg.Pool({
+                ...this.config,
+                connectionTimeoutMillis: 5000,
+            });
+
+            pool.on('error', (err) => {
+                console.error(`Postgres pool error ${err.message}`);
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:178]: I didn't trigger this error, but added for completion

**Review by owlas — APPROVED** [review]

**Pull request merged**